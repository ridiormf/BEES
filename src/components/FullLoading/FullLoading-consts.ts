export const FULL_LOADING = {
  VALUES: {
    MAX_Z_INDEX: 2000,
    MIN_Z_INDEX: -1,
  },
};
