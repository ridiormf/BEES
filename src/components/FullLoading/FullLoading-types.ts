export interface FullLoadingProps {
  visible?: boolean;
}
