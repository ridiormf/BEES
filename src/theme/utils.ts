type Space = number | string;

const utils = {
  marginHorizontal: (value: Space) => ({
    marginLeft: value,
    marginRight: value,
  }),
  marginVertical: (value: Space) => ({
    marginTop: value,
    marginBottom: value,
  }),
  paddingHorizontal: (value: Space) => ({
    paddingLeft: value,
    paddingRight: value,
  }),
  paddingVertical: (value: Space) => ({
    paddingTop: value,
    paddingBottom: value,
  }),
};

export default utils;
