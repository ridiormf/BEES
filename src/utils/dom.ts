export const getBody = () => {
  return document.getElementsByTagName('body')[0];
};
