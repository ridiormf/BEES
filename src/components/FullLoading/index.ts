import FullLoading from './FullLoading';
import type { FullLoadingProps } from './FullLoading-types';

export default FullLoading;
export type { FullLoadingProps };
