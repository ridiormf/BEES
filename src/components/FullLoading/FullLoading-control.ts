import { CSS } from '@stitches/react';
import React from 'react';
import { useConstructor, useDidUpdate } from '../../utils/hooks';
import { FULL_LOADING } from './FullLoading-consts';
import { FullLoadingProps } from './FullLoading-types';

export const useFullLoadingControl = ({ visible }: FullLoadingProps) => {
  const initialContainerCSS = useConstructor<CSS>(() => {
    if (!visible)
      return { opacity: 0, zIndex: FULL_LOADING.VALUES.MIN_Z_INDEX };
    return { opacity: 1, zIndex: FULL_LOADING.VALUES.MAX_Z_INDEX };
  });

  const [containerCSS, setContainerCSS] = React.useState(initialContainerCSS);

  useDidUpdate(() => {
    setContainerCSS(visible ? { opacity: 1 } : { opacity: 0 });
  }, [visible]);

  let containerClassName = '';
  if (!containerCSS!.zIndex) {
    containerClassName = visible ? 'show' : 'hide';
  }

  return {
    state: {
      containerCSS,
      containerClassName,
    },
  };
};
