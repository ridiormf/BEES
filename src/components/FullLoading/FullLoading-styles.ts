import { CSS, keyframes } from '@stitches/react';
import theme from '../../theme';
import { FULL_LOADING } from './FullLoading-consts';

const opacityUp = keyframes({
  '0%': { zIndex: FULL_LOADING.VALUES.MAX_Z_INDEX },
  '100%': { zIndex: FULL_LOADING.VALUES.MAX_Z_INDEX },
});

const opacityDown = keyframes({
  '0%': { zIndex: FULL_LOADING.VALUES.MAX_Z_INDEX },
  '100%': { zIndex: FULL_LOADING.VALUES.MIN_Z_INDEX },
});

const Container = theme.styled('div', {
  width: '100%',
  height: '100%',
  backgroundColor: '$primaryBackground',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  position: 'fixed',
  transition: '500ms',
  '&.show': {
    animation: `${opacityUp} 500ms forwards`,
  },
  '&.hide': {
    animation: `${opacityDown} 500ms forwards`,
  },
});

const scaleUp = keyframes({
  '0%': { transform: 'scale(1)' },
  '60%': { transform: 'scale(1.3)' },
  '100%': { transform: 'scale(1)' },
});

const imageCSS: CSS = {
  width: '10%',
  animation: `${scaleUp} 2000ms infinite`,
  '@lgDown': {
    width: '20%',
  },

  '@smDown': {
    width: '40%',
  },
};

export const fullLoadingStyles = {
  styled: {
    Container,
  },
  css: {
    image: imageCSS,
  },
};
