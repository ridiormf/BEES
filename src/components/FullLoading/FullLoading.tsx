import React from 'react';
import { createPortal } from 'react-dom';

import { fullLoadingStyles } from './FullLoading-styles';
import { FullLoadingProps } from './FullLoading-types';

import Image from '../Image';

import BeesImage from '../../assets/png/bees.png';
import { useFullLoadingControl } from './FullLoading-control';
import { getBody } from '../../utils/dom';

const {
  styled: { Container },
  css,
} = fullLoadingStyles;

const FullLoading: React.FC<FullLoadingProps> = ({ visible }) => {
  const {
    state: { containerCSS, containerClassName },
  } = useFullLoadingControl({ visible });
  return createPortal(
    <Container css={containerCSS!} className={containerClassName}>
      <Image src={BeesImage} css={css.image} alt="Bee image" />
    </Container>,
    getBody(),
  );
};

export default FullLoading;
