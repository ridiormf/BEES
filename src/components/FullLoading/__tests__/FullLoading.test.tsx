import { render } from '@testing-library/react';
import { Wrapper } from '../../../__tests__/utils';
import FullLoading from '../FullLoading';

describe('components ->', () => {
  describe('FullLoading ->', () => {
    describe('snapshots ->', () => {
      it('Should match snapshot', () => {
        const { container } = render(<FullLoading />, {
          wrapper: Wrapper,
        });
        expect(container).toMatchSnapshot();
      });
    });
  });
});
