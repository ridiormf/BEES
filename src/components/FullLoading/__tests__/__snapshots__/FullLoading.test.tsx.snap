// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components -> FullLoading -> snapshots -> Should match snapshot 1`] = `<div />`;
